spring-kafka/src/test/java/org/springframework/kafka/support/micrometer/MicrometerMetricsTests.java

```diff
@@ -0,0 +1,210 @@
+
+		Timer timer = meterRegistry.find("spring.kafka.listener")
+				.tags("name", "metricsTest-0")
+				.tag("result", "failure")
+				.timer();
+
+		assertThat(timer).isNotNull();
+		assertThat(timer.getId().getTag("exception"))
+				.isEqualTo("IllegalStateException");
+	}
+
+	@Test
+	void verifyMetricsWithObservation(@Autowired ObservationListener observationListener,
+			@Autowired MeterRegistry meterRegistry,
+			@Autowired KafkaTemplate<Integer, String> template)
+			throws Exception {
+
+		template.send(METRICS_TEST_TOPIC, "test").get(10, TimeUnit.SECONDS);
+		assertThat(observationListener.latch.await(10, TimeUnit.SECONDS)).isTrue();
+
+		Timer timer = meterRegistry.find("spring.kafka.listener")
+				.tag("spring.kafka.listener.id", "observationTest-0")
+				.tag("error", "IllegalStateException")
+				.timer();
+
+		assertThat(timer).isNotNull();
+	}
+
+	@Configuration
+	@EnableKafka
+	static class Config {
+
+		@Bean
+		KafkaAdmin admin(EmbeddedKafkaBroker broker) {
+			return new KafkaAdmin(Map.of(
+					AdminClientConfig.BOOTSTRAP_SERVERS_CONFIG,
+					broker.getBrokersAsString()));
+		}
+
+		@Bean
+		ProducerFactory<Integer, String> producerFactory(EmbeddedKafkaBroker broker) {
+			return new DefaultKafkaProducerFactory<>(
+					KafkaTestUtils.producerProps(broker));
+		}
+
+		@Bean
+		ConsumerFactory<Integer, String> consumerFactory(EmbeddedKafkaBroker broker) {
+			return new DefaultKafkaConsumerFactory<>(
+					KafkaTestUtils.consumerProps("metrics", "false", broker));
+		}
+
+		@Bean
+		KafkaTemplate<Integer, String> template(ProducerFactory<Integer, String> pf) {
+			return new KafkaTemplate<>(pf);
+		}
+
+		@Bean
+		ConcurrentKafkaListenerContainerFactory<Integer, String> kafkaListenerContainerFactory(
+				ConsumerFactory<Integer, String> cf) {
+			ConcurrentKafkaListenerContainerFactory<Integer, String> factory =
+					new ConcurrentKafkaListenerContainerFactory<>();
+			factory.setConsumerFactory(cf);
+			factory.getContainerProperties().setMicrometerEnabled(true);
+			factory.getContainerProperties().setObservationEnabled(false);
+			return factory;
+		}
+
+		@Bean
+		ConcurrentKafkaListenerContainerFactory<Integer, String> observationListenerContainerFactory(
+				ConsumerFactory<Integer, String> cf, ObservationRegistry observationRegistry) {
+			ConcurrentKafkaListenerContainerFactory<Integer, String> factory =
+					new ConcurrentKafkaListenerContainerFactory<>();
+			factory.setConsumerFactory(cf);
+			factory.getContainerProperties().setObservationEnabled(true);
+			factory.getContainerProperties().setObservationRegistry(observationRegistry);
+			return factory;
+		}
+
+		@Bean
+		MetricsListener metricsListener() {
+			return new MetricsListener();
+		}
+
+		@Bean
+		MeterRegistry meterRegistry() {
+			return new SimpleMeterRegistry();
+		}
+
+		@Bean
+		ObservationListener observationListener() {
+			return new ObservationListener();
+		}
+
+		@Bean
+		SimpleTracer simpleTracer() {
```

Comment:
Do we use this bean somewhere?

Reply:
sorry - copy error from the other test :)